SWIPs/swip-multiple_payment_processing_support.md

```diff
@@ -55,7 +55,7 @@ At a high level a payment module is responsible for:
 * Referencing a type, version and base currency 
 * Optionally exposing other methods such as querying balances, topping up balances or sending payments (outside of Swarm). 
 
-Nodes can specify their preference for both payment module, as well as price oracle in a list in a configuration file. The preferences are normalized and weighed.
+Nodes can specify their preference for both payment module, as well as price oracle in a list in a configuration file. The preferences are normalized and weighted.
 For any preference list, the chosen option will be the option which has the highest cumulative preference. The preference list has three dimensions, which will be resolved from high to low:
 
 1. Currency to use
```

Comment:
I guess it's implicit but if for example nodes can't agree in the last dimension (payment method), then the fall back will be applied to all three dimensions ,right?

Reply:
That's a good question. The outcome should anyways be that nodes should always be able to connect and that they should be able to make other choices when it is beneficial for both. I think it is not needed that the fallback is applied to all three dimensions when there is no agreement on one of the dimensions. Imagine two preference lists:
```
Node A:
1. Bitcoin: Very strong preference
2. Oracle A: Neutral
3. Lightning: Preferably not, ChequebookOnBitcoin: Strong preference.

Node B
1. Bitcoin: Very strong preference
2. Oracle A: Neutral
3. Lightning: Strong preference, ChequebookOnBitcoin: preferably not.
```
A and B don't come to an agreement on the 3rd dimension, but I think that they should both still use Bitcoin for the 1st dimentions. In such a case, it is not desirable that the fallback will be used for all three dimensions.

Reply:
Well firstly, in your example you introduce new notions which are not present in the SWIP so far: "Very strong preference", "Neutral", etc.. Where is that defined? The SWIP only says:
```
Dimension: 
1. Currency to use
2. Price Oracle
3. Payment method
```

Thus I would see your examples as 
```
Node A: 
1. Bitcoin
2. Oracle address
3. Lightning
```
If they can't agree on 3, it is AFAIU not possible to evaluate how they can proceed with Bitcoin, as the payment method is undefined and thus they need to fallback to default for everything.

Reply:
You can't have `Oracle A: neutral`, I understood this to be the address, so there should be a valid address there. Otherwise the whole section needs clarification, as I clearly would have misunderstood